src/components/Button/Button.tsx

```diff
@@ -1,10 +1,45 @@
 import React, { ButtonHTMLAttributes } from "react";
+import ButtonOutline from "./ButtonOutline";
+import ButtonText from "./ButtonText";
+import { ThemeProvider } from "styled-components";
+import ButtonMode from './ButtonMode';
 
-function Button(props: ButtonHTMLAttributes<HTMLButtonElement>) {
+interface ButtonTheme {
```

Comment:
I think we can technically extend the `ButtonHTMLAttributes<HTMLButtonElement>` interface here (and extract the `ButtonTheme` interface outside of the component definition) rather than specify the union type.